decidim-core/app/services/decidim/onboarding_manager.rb

```diff
@@ -0,0 +1,231 @@
+  #    completed, it will remove the pending action from the User#extended_data column and redirect
+  #    the user to the intended action.
+  #
+  class OnboardingManager
+    DATA_KEY = "onboarding"
+
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    # Checks if the onboarding data has an action and a model.
+    #
+    # Returns a boolean
+    def valid?
+      return if action.blank?
+
+      permissions_holder.present?
+    end
+
+    # Returns the action to be performed.
+    #
+    # Returns a string
+    def action
+      onboarding_action
+    end
+
+    # Returns the translation of the action to be performed if found, action
+    # elsewhere
+    #
+    # Returns a string
+    def action_text
+      @action_text ||= if component && I18n.exists?("#{component.manifest.name}.actions.#{action}", scope: "decidim.components")
+                         I18n.t("#{component.manifest.name}.actions.#{action}", scope: "decidim.components")
+                       elsif permissions_holder.respond_to?(:resource_manifest) &&
+                             I18n.exists?("#{permissions_holder.resource_manifest.name}.actions.#{action}", scope: "decidim.resources")
+                         I18n.t("#{permissions_holder.resource_manifest.name}.actions.#{action}", scope: "decidim.resources")
+                       else
+                         action
+                       end
+    end
+
+    # Checks if there is any pending action for the user.
+    #
+    # Returns a boolean
+    def pending_action?
+      onboarding_data.present?
+    end
+
+    # Checks if the user has completed all the required authorizations.
+    #
+    # active_authorization_methods - A list of the active authorization methods for the user.
+    #
+    # Returns a boolean
+    def finished_verifications?(authorization_methods = active_authorization_methods)
+      (authorization_handlers - authorization_methods).empty?
+    end
+
+    # Returns the model related to the action in the onboarding process.
+    #
+    # Returns an ActiveRecord model
+    def model
+      return if onboarding_model.blank?
+
+      @model ||= GlobalID::Locator.locate(onboarding_model)
+    end
+
+    # Returns the permissions_holder if present related to the action in the
+    # onboarding process.
+    #
+    # Returns an ActiveRecord model
+    def permissions_holder
+      return model if onboarding_permissions_holder.blank?
+
+      @permissions_holder ||= GlobalID::Locator.locate(onboarding_permissions_holder)
+    end
+
+    # Returns the model name related to the action in the onboarding process.
+    #
+    # Returns a string
+    def model_name
+      return unless valid?
+
+      @model_name ||= (model.presence || permissions_holder).class.model_name
+    end
+
+    # Returns the resource title associated to the action. If the model is defined
+    # its title is used, if not the permissions holder title
+    #
+    # Returns a Hash
```

Comment:
Hash or String?

Reply:
It's a translations Hash in general, but it can be a String in some cases. The returned value of this method is processed with `decidim_sanitize_translated`. I have expanded the comment